From acc40ab043befe6251168d0732bc2ed535a491e2 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Thu, 28 Nov 2024 14:19:18 +0800
Subject: [PATCH 149/149] mtk: hostapd: reset bss->scan_link after getting
 survey

The bss->scan_link is used for AP MLD to record the link that
requested the scan. Subsequently, the survey results can be passed
to the "scan_link".
However, not resetting the bss->scan_liunk results in subsequent scans
always pass the result to that "scan_link".

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/drivers/driver_nl80211.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index c841ee3fc..5d08d6f1b 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -10697,6 +10697,7 @@ static int wpa_driver_nl80211_get_survey(void *priv, unsigned int freq,
 		wpa_supplicant_event(ctx, EVENT_SURVEY, &data);
 
 	clean_survey_results(survey_results);
+	bss->scan_link = NULL;
 	return err;
 }
 
-- 
2.45.2

